test/test_c_parser.py

```diff
@@ -303,3 +305,31 @@ def test_syntax_error_for_file(text: str, tmp_path: PurePath) -> None:
         tb = traceback.format_exc()
     assert 'some_file.py", line 1' in tb
     assert f"{text}\n        ^" in tb
+
+
+def test_metas(tmp_path: PurePath) -> None:
+    grammar_source = f"""
+    @header '''\
+        {textwrap.indent(EXTENSION_PREFIX, "    ")}
+        expr_ty identity_3(expr_ty x);
+        expr_ty identity_1(expr_ty x) {{
+            return identity_3(x);
+        }}
+    '''
+    @subheader '''
+        expr_ty identity(expr_ty x) {{
+            return identity_1(x);
+        }}
+    '''
+    @trailer '''
+        expr_ty identity_3(expr_ty x) {{
+            return x;
+        }}
+        {textwrap.indent(EXTENSION_SUFFIX, "    ")}
+    '''
+    start: expr+ NEWLINE? ENDMARKER
+    expr: x=NAME {{identity(x)}}
+    """
+    grammar = parse_string(grammar_source, GrammarParser)
+    extension = generate_parser_c_extension(grammar, tmp_path)
+    extension.parse_string("a b c")
```

Comment:
Why not write a test that tests exactly what is being generated, rather than trying to compile and run it? If necessary, allow better dependency injection to allow such a test to be written.

Reply:
> Why not write a test that tests exactly what is being generated, rather than trying to compile and run it?

Originally, I wanted to reuse as much machinery as possible, but I agree that testing the contents of the generated would make the test simpler.